packages/core/helper-plugin/src/utils/once.ts

```diff
@@ -1,17 +1,18 @@
 export const PREFIX = '[@strapi/helper-plugin]:';
 
-/**
- * @type {<TFunc extends (...args: any) => any>(fn: TFunc) => (...args: any[]) => void}
- */
-export const once = (fn) => {
+export const once: <TFunc extends (...args: any) => any>(fn: TFunc) => (...args: any[]) => void = <
+  TFunc extends (...args: any[]) => any
+>(
+  fn: TFunc
+): ((...args: any[]) => void) => {
```

Comment:
this is super hard to read why have you doubled up on the declaration? :)

Reply:
Would it be more clear?

```
type AnyFunction = (...args: any[]) => any;

export const once = <TFunc extends AnyFunction>(fn: TFunc): AnyFunction => {
```

Reply:
i'd probably just do `export const once = <TFunc extends (...args: any) => any>(fn: TFunc) => {}` because we can infer the return type?

Reply:
@joshuaellis I fully agree! I have applied the suggestion. Thank you so much for the review!